### ISSUE TITLE
[Snackbar] Can't style Snackbar action text unless also styling text label

### ISSUE THREAD

**Issue by moopat** [material-components/material-components-android#2134]
**Description:** I am trying to customize the text color of the Snackbar's action [according to the documentation](https://material.io/components/snackbars/android#theming-snackbars). I was unable to adjust the color of the action text. A quick look at [this changeset](https://github.com/material-components/material-components-android/commit/b99709200e42366bec214c1e862f5f69ba22d516#diff-e19f85ce9e235bfd327e6c94cd6bebbdcbc074b6943cebfdd29d4cff9167605b) made me aware that in order to customize the action, I need to customize the Snackbar's text label as well.

**Expected behavior:** I expect to be able to change the action on its own, without providing a style for the text label. If the current implementation is intended behaviour it should be part of the documentation.

**Source code:** 

I tried setting a custom `snackbarButtonStyle`, which didn't work:

```xml
<style name="AppTheme" parent="Theme.MaterialComponents.Light.Bridge">
    <!-- [...] -->
    <item name="snackbarButtonStyle">@style/Widget.App.SnackbarButton</item>
</style>

<style name="Widget.App.SnackbarButton" parent="Widget.MaterialComponents.Button.TextButton.Snackbar">
    <item name="android:textColor">@color/white</item>
</style>
```
When I added a `snackbarTextViewStyle` (which just references the default style) it worked:

```xml
<style name="AppTheme" parent="Theme.MaterialComponents.Light.Bridge">
    <!-- [...] -->
    <item name="snackbarButtonStyle">@style/Widget.App.SnackbarButton</item>
    <item name="snackbarTextViewStyle">@style/Widget.MaterialComponents.Snackbar.TextView</item>
</style>

<style name="Widget.App.SnackbarButton" parent="Widget.MaterialComponents.Button.TextButton.Snackbar">
    <item name="android:textColor">@color/white</item>
</style>
```

**Android API version:** 30

**Material Library version:** 1.3.0

**Device:** Any device, but Pixel 4

To help us triage faster, please check to make sure you are using the [latest version](https://github.com/material-components/material-components-android/releases) of the library.

We also happily accept [pull requests](https://github.com/material-components/material-components-android/pulls).

**Comment by thsaravana** [comment]
Yes, this should be mentioned in the documentation if this is an intended change. I had to go through the code to figure out why `snackbarButtonStyle` was never picked up.

The [documentation](https://material.io/components/snackbars/android#anatomy-and-key-properties) states the default value of `snackbarButtonStyle` is `Widget.MaterialComponents.Button.TextButton.Snackbar` and `snackbarTextViewStyle` is `Widget.MaterialComponents.Snackbar.TextView`. But looking at the code, even this is not true.
Ideally, the default values should be applied and we should have the control to override _any style, if we want to_.
